### PR TITLE
Create BaseIssuer type to implement Match for all Issuers

### PR DESCRIPTION
**Pull request by priyawadhwa**
Also includes support for regex's in Match for MetaIssuer types

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
looks nice

**Review by haydentherapper — COMMENTED** [review]
Looks good!

File: pkg/identity/base/issuer.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"strings"
+
+	"github.com/sigstore/fulcio/pkg/identity"
+)
+
+type baseIssuer struct {
+	issuerURL string
+}
+
+func Issuer(issuerURL string) identity.Issuer {
```
Comment (haydentherapper) [inline pkg/identity/base/issuer.go:30]: Do we need a public Issuer function? If this is only used for tests, can we make it private or just initialize with the `baseIssuer` type?

Comment (priyawadhwa) [reply pkg/identity/base/issuer.go:30]: all of the other issuer types call it!

**Pull request merged**